### DISCUSSION TITLE
Table: single-selectable row

### DISCUSSION THREAD

**Discussion by bentleyvk** [iTwin/iTwinUI-react#508] [General]
More and more people ask about single-selectable rows inside Table and we always give them workarounds. Maybe we need to think of a standard for this, whether we want to implement it. @FlyersPh9 your opinion would be great here too.

**Comment by mayank99** [comment]
I think the problem here could be easily solved by making the checkbox column optional. When checkboxes are shown, multi selection is the common pattern. When single select is needed, checkboxes simply don't make sense. 🤷‍♀️

This way we also decouple the "selected" with "active". It's a small but important difference (similar to checkbox vs toggle switch). To me:
- checkboxes imply that there is no immediate effect (there is still time to review before continuing to the next step).
- absence of checkboxes implies that the action has immediate effect (click on a row -> it opens a new overlay or changes what is displayed in viewer)

**Reply by FlyersPh9** [reply]
At first I did not understand the request, but I chatted with @bentleyvk and I understand now.  @mayank99, your 2 bullet points make a lot of sense.  I support making the checkbox column optional with the rules of "when checkbox is present & an item is selected, there is no immediate effect.  when checkboxes are missing, effect is immediate".

@VirginieGaudreault will update the table standards to reflect these changes.

**Comment by bentleyvk** [comment]
Created an issue https://github.com/iTwin/iTwinUI-react/issues/509
